doc/api/corepack.md

```diff
@@ -0,0 +1,108 @@
+
+> Stability: 1 - Experimental
+
+_[Corepack][]_ is an experimental tool to help with
+managing versions of your package managers. It exposes binary proxies for
+each [supported package manager][] that, when called, will identify whatever
+package manager is configured for the current project, transparently install
+it if needed, and finally run it without requiring explicit user interactions.
+
+This feature simplifies two core workflows:
+
+* It eases new contributor onboarding, since they won't have to follow
+  system-specific installation processes anymore just to have the package
+  manager you want them to.
+
+* It allows you to ensure that everyone in your team will use exactly the
+  package manager version you intend them to, without them having to
+  manually synchronize it each time you need to make an update.
+
+## Workflows
+
+### Enabling the feature
+
+Due to its experimental status, Corepack currently needs to be explicitly
+enabled to have any effect. To do that simply run [`corepack enable`][], which
+will set up the symlinks in your environment, next to the `node` binary
+(and overwrite the existing symlinks if necessary).
+
+From this point forward, any call to the [supported binaries][] will work
+without further setup. Should you experience a problem, just run
+[`corepack disable`][] to remove the proxies from your system (and consider
+opening up an issue on the [Corepack repository][] to let us know).
+
+### Configuring a package
+
+The Corepack proxies will find the closest [`package.json`][] file in your
+current directory hierarchy to extract its [`"packageManager"`][] property.
+
+If the value corresponds to a [supported package manager][], Corepack will make
+sure that all calls to the relevant binaries are run against the requested
+version, downloading it on demand if needed, and aborting if it cannot be
+successfully retrieved.
+
+### Upgrading the global versions
+
+When running outside of an existing project (for example when running
+`yarn init`), Corepack will by default use predefined versions roughly
+corresponding to the latest stable releases from each tool. Those versions can
+be easily overriden by running the [`corepack prepare`][] command along with the
+package manager version you wish to set:
+
+```bash
+corepack prepare yarn@x.y.z --activate
+```
+
+### Offline workflow
+
+Many production environments don't have network access. Since Corepack
+usually downloads the package manager releases straight from their registries,
+it can conflict with such environments. To avoid that happening, call the
+[`corepack prepare`][] command while you still have network access (typically at
+the same time you're preparing your deploy image). This will ensure that the
+required package managers are available even without network access.
+
+The `prepare` command has [various flags][], consult the detailed
+[Corepack documentation][] for more information on the matter.
+
+## Supported package managers
```

Comment:
I feel like there should be a paragraph warning that Corepack has some virtual support for `npm` (it can download it and accepts it in the "packageManager" field), but that has no effect on the user.

Reply:
Also, it doesn't prevent from using `npm` in a project configured for `yarn` or `pnpm`.

Reply:
This last part is intentional - this PR doesn't change anything for npm users, both for compatibility and "jurisdiction" reasons. I think they and their users would benefit from being integrated with Corepack too, but it needs to come from them.

Reply:
I understand it's intentional, just saying that it should be stated in the documentation.

Reply:
I've updated the documentation to clarify the text according your two remarks; does it look better?